src/kinetics/InterfaceKinetics.cpp

```diff
@@ -767,11 +767,14 @@ doublereal InterfaceKinetics::electrochem_beta(size_t irxn) const
     return 0.0;
 }
 
-void InterfaceKinetics::advanceCoverages(doublereal tstep)
+void InterfaceKinetics::advanceCoverages(doublereal tstep, doublereal rtol,
+                                         doublereal atol, doublereal maxStepSize,
+                                         size_t maxSteps, size_t maxErrTestFails)
 {
     if (m_integrator == 0) {
         vector<InterfaceKinetics*> k{this};
-        m_integrator = new ImplicitSurfChem(k);
+        m_integrator = new ImplicitSurfChem(k, rtol, atol, maxStepSize, maxSteps,
+                                            maxErrTestFails);
```

Comment:
This probably isn't a big issue given the typical use cases, but as written here, changes to the integrator parameters on subsequent calls to `advanceCoverages` will have no effect. This could be fixed by moving the setting of these parameters to a different method that can be called either on a new or existing `ImplicitSurfChem` object, rather than requiring them to be constructor arguments.